src/plugins/vis_type_vega/public/data_model/vega_parser.ts

```diff
@@ -116,20 +161,20 @@ export class VegaParser {
     // eslint-disable-next-line import/namespace
```

Comment:
There are a lot of `this.vlspec?.` and `this.spec?.` in this function  - can we check whether they are set once and just exit early if that's not the case (like in the other comment)? Seems easier than using the safe accessors everywhere.